fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapper.kt

```diff
@@ -39,6 +39,7 @@ private fun GeckoProductAnalysis.toProductReview(): ProductReviewState =
                 productId = productId!!,
                 reviewGrade = mappedGrade,
                 needsAnalysis = needsAnalysis,
+                reanalysisInProgress = false,
```

Comment:
this is where I'm not sure how the reanalysis API will affect our state machine model, so this is placeholder just to get the code compiling and have the UI demoable

Reply:
The `needsAnalysis` Boolean would be converted to: 
```
enum class AnalysisStatus {
  // we have this covered with the start reanalysis card
  NEEDS_ANALYSIS, 
  
  // that's this loading 
  REANALYZING, 

  // no need to show anything, the ideal case.
  UP_TO_DATE, 
  
  // the green analysis is completed card.
  COMPLETED
}
```
as only one of them will be displayed for the case of `AnalysisPresent`.

This will be handled in this [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1853299). I think in this PR, we can add the loading card and not integrate that in the `ProductAnalysis` composable. That way we can decouple this PR from the state update which is decoupled from the API integration in GV.

Reply:
that works for me